Scripting API documentation (#14725)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/scripting/registercontentscripts/index.md

---
title: scripting.registerContentScripts()
slug: Mozilla/Add-ons/WebExtensions/API/scripting/registerContentScripts
tags:
  - API
  - Add-ons
  - Extensions
  - Method
  - Reference
  - WebExtensions
  - registerContentScripts
  - scripting
browser-compat: webextensions.api.scripting.registerContentScripts
---
{{AddonSidebar()}}

Registers one or more content scripts. 

> **Note:** This method is available in Manifest V3 or higher.

> This is an asynchronous function that returns a [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).

Reviewer: ```suggestion
 To use this API you must have the `"scripting"` [permission](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions) and permission for the page's URL, either explicitly as a [host permission](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#host_permissions) or using the [activeTab permission](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#activetab_permission).
 
This is an asynchronous function that returns a [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
```

## Syntax

```js
await browser.scripting.registerContentScripts(
  scripts         // array
)
```

### Parameters

- `scripts`

>   - : `array` of {{WebExtAPIRef("scripting.RegisteredContentScript")}}. A list of scripts to register. 

Reviewer: is this correct? (no name before the colon)

Author: Perhaps I'm missing something, but this is following the standard formatting

### Return value

> A [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that fulfills with an array of {{WebExtAPIRef("scripting.RegisteredContentScript")}}. If there are errors during script parsing and file validation, or if the IDs specified exist, no scripts are registered and the promise is rejected.

Reviewer: ```suggestion
A [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that fulfills with an array of {{WebExtAPIRef("scripting.RegisteredContentScript")}}. If there are errors during script parsing and file validation, or if the IDs specified do not exist, no scripts are registered and the promise is rejected.
```

## Examples

> This example registers a script file called `script.js`:

Reviewer: ```suggestion
This example registers a content script that will inject the file `"script.js"`:
```

```js

> const aScript = {
>   id: "a-script",
>   js: ["script.js"],
>   matches: ["https://example.com/*"],
>   persistAcrossSessions: false,
> };
> 
> await browser.scripting.registerContentScripts([aScript]);

Reviewer: ```suggestion
const aScript = {
  id: "a-script",
  js: ["script.js"],
  matches: ["https://example.com/*"],
};

await browser.scripting.registerContentScripts([aScript]);
```

---
title: scripting.registerContentScripts()
slug: Mozilla/Add-ons/WebExtensions/API/scripting/registerContentScripts
tags:
  - API
  - Add-ons
  - Extensions
  - Method
  - Non-standard
  - Reference
  - WebExtensions
  - registerContentScripts
  - scripting
browser-compat: webextensions.api.scripting.registerContentScripts
---
{{AddonSidebar()}}

> Registers one or more content scripts. If there are errors during script parsing and file validation, or if the IDs specified exist, no scripts are registered.

Author: Moved to returned value section
